src/mono/mono/metadata/cominterop.c

```diff
@@ -2998,12 +3012,19 @@ mono_ptr_to_bstr (const gunichar2* ptr, int slen)
 #ifndef DISABLE_COM
 	if (com_provider == MONO_COM_DEFAULT) {
 #endif
+#ifdef ENABLE_NETCORE
+		mono_bstr const s = (mono_bstr)mono_bstr_alloc ((slen + 1) * sizeof (gunichar2));
+		if (s == NULL)
+			return NULL;
+#else // In Mono, historically BSTR was allocated with a guaranteed size prefix of 4 bytes regardless of platform
 		/* allocate len + 1 utf16 characters plus 4 byte integer for length*/
 		guint32 * const ret = (guint32 *)g_malloc ((slen + 1) * sizeof (gunichar2) + sizeof (guint32));
 		if (ret == NULL)
 			return NULL;
 		mono_bstr const s = (mono_bstr)(ret + 1);
-		*ret = slen * sizeof (gunichar2);
+#endif
+		// Set the length
+		*((guint32 *)s - 1) = slen * sizeof (gunichar2);
```

Comment:
I may be wrong here but, in case of 64-bit pointers, `*((guint32 *)s - 2)` may not be getting zeroed

Reply:
It looks like this problem is still there in the new set_length function

Reply:
Oh I see the switch to `g_malloc0` now